spec/features/consumer/caching/shops_caching_spec.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+feature "Shops caching", js: true, caching: true do
+  let!(:distributor) { create(:distributor_enterprise, with_payment_and_shipping: true, is_primary_producer: true) }
+  let!(:order_cycle) { create(:open_order_cycle, distributors: [distributor], coordinator: distributor) }
+
+  describe "caching enterprises AMS data" do
+    it "caches data for all enterprises, with the provided options" do
+      visit shops_path
+
+      key, options = CacheService::FragmentCaching.ams_shops
+      expect_cached "views/#{key}", options
+    end
+
+    it "keeps data cached for a short time on subsequent requests" do
+      # One minute ago...
+      Timecop.travel(Time.zone.now - 1.minute) do
```

Comment:
```suggestion
      Timecop.travel(1.minute.ago) do
```

I reckon that `ago` is a terrible language design on Rails' part but very convenient.

Reply:
But would it be better to use `SHOPS_EXPIRY` here? One minute is a magic number. Maybe something like `SHOPS_EXPIRY + 2 * Capybara.default_max_wait_time`? The `2` is the number of page visits here.

Actually, if the server is really slow, `expect` can wait up to 30 seconds and then the spec fails. Flaky. If we are sure that this page is fast, we can reduce Capybara's wait time here to make sure that we are within the cache time. Or, and I like this idea more, we stub the caching time to be a multiple of Capybara's wait time.

Reply:
`10.minutes.ago` should be safe for any theoretical flakyness, right? It's still a magic number, but we just need it to be longer than 30-150 seconds.

Reply:
It hasn't actually failed on a slow machine yet, right?

Reply:
It's a really good point Maikel!
I think using 10 or 30 minutes is enough here. I'd add a comment explaining the magic: basically a number bigger than cache time plus any capybara wait time.